Champions/Taric/R.cs

```diff
@@ -0,0 +1,88 @@
+using System.Linq;
+using GameServerCore;
+using GameServerCore.Domain;
+using GameServerCore.Domain.GameObjects;
+using GameServerCore.Enums;
+using static LeagueSandbox.GameServer.API.ApiFunctionManager;
+using LeagueSandbox.GameServer.Scripting.CSharp;
+using LeagueSandbox.GameServer.GameObjects.AttackableUnits.AI;
+
+namespace Spells
+{
+    public class TaricHammerSmash : IGameScript
+    {
+        public void OnActivate(IChampion owner)
+        {
+        }
+
+        public void OnDeactivate(IChampion owner)
+        {
+        }
+
+        public void OnStartCasting(IChampion owner, ISpell spell, IAttackableUnit target)
+        {
+            
+        }
+
+        public void OnFinishCasting(IChampion owner, ISpell spell, IAttackableUnit target)
+        {
+            var p1 = AddParticleTarget(owner, "TaricHammerSmash_nova.troy", owner);
+            var p2 = AddParticleTarget(owner, "TaricHammerSmash_shatter.troy", owner);
+            var hasbuff = owner.HasBuffGameScriptActive("Radiance", "Radiance");
+            var ap = owner.Stats.AbilityPower.Total * 0.5f;
+            var damage = 50 + spell.Level * 100 + ap;
+
+            foreach (var enemyTarget in GetUnitsInRange(owner, 375, true)
+                .Where(x => x.Team == CustomConvert.GetEnemyTeam(owner.Team)))
+            {
+                if (enemyTarget is IChampion || enemyTarget is IMinion && owner.Team != enemyTarget.Team)

```

Comment:
Add parentheses to clarify (even if technically you don't have to).

Reply:
I will change it to IAttackableUnit, but it will cause to crash server, because IAttackableUnit included "Turrets/Inhibitors/Nexus".